Src/FluentAssertions.Net40/Xml/XmlReaderValidator.cs

```diff
@@ -0,0 +1,239 @@
+﻿using FluentAssertions.Execution;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+
+namespace FluentAssertions.Xml
+{
+    class XmlReaderValidator
+    {
+        AssertionScope assertion;
+        XmlReader subjectReader, expectedReader;
+
+        string CurrentLocation
+        {
+            get { return "/" + string.Join("/", locationStack.Reverse()); }
+        }
+
+        Stack<string> locationStack = new Stack<string>();
+
+        public XmlReaderValidator(XmlReader subjectReader, XmlReader expectedReader, string because, object[] reasonArgs)
+        {
+            assertion = Execute.Assertion.BecauseOf(because, reasonArgs);
+        
+            this.subjectReader = subjectReader;
+            this.expectedReader = expectedReader;
+        }
+                
+        class ValidationResult
```

Comment:
Yes, that's why I use FA in the first place :-)

If I find the time, I'd love to write a better error aggregation, but I think that already as this is, it's a vast improvement for Xml work over the existing features.